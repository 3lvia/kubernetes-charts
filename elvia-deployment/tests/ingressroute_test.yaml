suite: test ingressroute
templates:
  - ingressroute.yaml
tests:
  - it: ingressroute snapshot should match
    set:
      image.tag: latest
      name: demo-api
      namespace: core
      microserviceType: webapi
      environment: dev
      replicaCount: 1
      resources:
        requests:
          cpu: 100m
          memory: 128Mi
        limits:
          cpu: 400m
          memory: 500Mi
      service:
        port: 80
        targetPort: 8080
      ingress:
        subdomain: core
        path: "/demo-api"
    asserts:
      - matchSnapshot: {}
