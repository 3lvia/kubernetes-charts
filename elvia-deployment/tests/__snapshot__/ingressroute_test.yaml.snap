ingressroute snapshot should match:
  1: |
    apiVersion: traefik.io/v1alpha1
    kind: IngressRoute
    metadata:
      name: demo-api
      namespace: core
    spec:
      routes:
        - kind: Rule
          match: Host(`core.dev-elvia.io`) && PathPrefix(`/demo-api`)
          middlewares:
            - name: demo-api-stripprefix
            - name: hsts-header
              namespace: dns
            - name: remove-metrics
              namespace: dns
          services:
            - name: demo-api
              port: 80
  2: |
    apiVersion: traefik.io/v1alpha1
    kind: Middleware
    metadata:
      name: demo-api-stripprefix
      namespace: core
    spec:
      stripPrefix:
        prefixes:
          - /demo-api
